Added a change to be able to ask for pull request.

SzabolcsBorbely opened the pull request.

Commit:
Added a change to be able to ask for pull request.

Commit:
Integration Test + Code update

Updated code and formatting as suggested in the comments.
Integration test passes.
TO-DO: Travis CI

Commit:
Added .travis.yml to project root for Travis CI

ZsoltFabok commented on src/main/java/fizzbuzz/FizzBuzz.java:
```diff
@@ -1,64 +1,56 @@
-public class FizzBuzz{
+public class FizzBuzz {
+    
+    final String FIZZ = "Fizz";
```
constants should always be static final

ZsoltFabok commented on src/main/java/fizzbuzz/FizzBuzz.java:
```diff
@@ -1,64 +1,56 @@
-	public List<String> generateSequence(Integer range_floor, Integer range_top){
-		List<String> result_list = new ArrayList<String>();
-		boolean divisible_by_3, divisible_by_5;
+    public boolean rangeIsValid(String range_floor, String range_top) {
```
Change this to the java8 Optional structure

ZsoltFabok commented on src/main/java/fizzbuzz/FizzBuzz.java:
```diff
@@ -1,64 +1,56 @@
+        try {
+            int floor = Integer.parseInt(range_floor);
+            int top = Integer.parseInt(range_top);
+            if (floor <= 0 || top <= 0)
```
missing {

ZsoltFabok commented on src/main/java/fizzbuzz/Main.java:
```diff
@@ -0,0 +1,27 @@
+    public static void testFizzBuzz(String[] args) {
+        if (args.length != 2) {
+            System.err.println("wrong arguments, try: <number> <number>");
+            return;
```
This is not really nice because the reader has to think a lot to figure out when and where this method exists. Moreover you have a void return type, therefore do not call `return;`. Refactor this part :-(

ZsoltFabok commented on src/main/java/fizzbuzz/Main.java:
```diff
@@ -0,0 +1,27 @@
+            System.err.println("wrong arguments, try: <number> <number>");
+            return;
+        }
+        for (int i=0, len=fizzbuzzsequence.size();i<len-1;i++) {
```
spaces

ZsoltFabok commented on src/main/java/fizzbuzz/Main.java:
```diff
@@ -0,0 +1,27 @@
+            return;
+        }
+        for (int i=0, len=fizzbuzzsequence.size();i<len-1;i++) {
+            System.out.print(fizzbuzzsequence.get(i)+" ");
```
spaces

ZsoltFabok commented on src/main/java/fizzbuzz/Main.java:
```diff
@@ -0,0 +1,27 @@
+        for (int i=0, len=fizzbuzzsequence.size();i<len-1;i++) {
+            System.out.print(fizzbuzzsequence.get(i)+" ");
+        }
+        System.out.print(fizzbuzzsequence.get(fizzbuzzsequence.size()-1)+System.lineSeparator());
```
== System.out.println()

ZsoltFabok commented on src/main/java/fizzbuzz/Main.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Main {
+    
+    public static void testFizzBuzz(String[] args) {
```
not really a good name.

ZsoltFabok commented on src/test/java/fizzbuzz/IntegrationTest.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Test
+    public void prints_error_message_for_wrong_arguments() {
+        Main.main(new String[] { "10", "buzz" });
```
Different structure than in the previous methods.

ZsoltFabok commented:
:-1:

Commit:
Major refactoring.

Changes were made to the code, in accordance with the previous comments
(excluding the Java 8 Optional<> part).
Test cases were created and modified as well, to fit the new class structure.
